src/tests/ResourceManagerTest.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <Corrade/Utility/Directory.h>
+#include <gtest/gtest.h>
+#include <string>
+
+#include "esp/assets/ResourceManager.h"
+#include "esp/gfx/Renderer.h"
+#include "esp/gfx/WindowlessContext.h"
+#include "esp/scene/SceneManager.h"
+
+#include "configure.h"
+
+namespace Cr = Corrade;
+
+using esp::assets::ResourceManager;
+using esp::scene::SceneManager;
+
+const std::string dataDir = Cr::Utility::Directory::join(SCENE_DATASETS, "../");
+
+TEST(ResourceManagerTest, JoinMesh) {
+  esp::gfx::WindowlessContext::uptr context_ =
+      esp::gfx::WindowlessContext::create_unique(0);
+
+  std::shared_ptr<esp::gfx::Renderer> renderer_ = esp::gfx::Renderer::create();
+
+  // must declare these in this order due to avoid deallocation errors
+  ResourceManager resourceManager;
+  SceneManager sceneManager_;
+
+  std::string boxFile = Cr::Utility::Directory::join(
+      dataDir, "test_assets/objects/transform_box.glb");
+
+  int sceneID = sceneManager_.initSceneGraph();
+  auto& sceneGraph = sceneManager_.getSceneGraph(sceneID);
+  esp::scene::SceneNode* navSceneNode = &sceneGraph.getRootNode().createChild();
+  const esp::assets::AssetInfo info = esp::assets::AssetInfo::fromPath(boxFile);
+  resourceManager.loadScene(info, navSceneNode, nullptr);
+
+  std::shared_ptr<esp::assets::MeshData> joinedBox =
+      resourceManager.joinMesh(boxFile);
+
+  // transform_box.glb is composed of 6 identical triangulated plane meshes
+  // transformed into a cube via a transform heirarchy. Combined, the resulting
+  // mesh should have 24 vertices and 36 indices with corners at the unit corner
+  // coordinates as defined in the ground truth vectors below.
+  int numVerts = joinedBox->vbo.size();
+  int numIndices = joinedBox->ibo.size();
+
+  ASSERT_EQ(numVerts, 24);
+  ASSERT_EQ(numIndices, 36);
+
+  std::vector<Magnum::Vector3> vertGroundTruth{
+      Magnum::Vector3{-1, 1, 1},  Magnum::Vector3{-1, -1, -1},
+      Magnum::Vector3{-1, 1, -1}, Magnum::Vector3{-1, -1, 1},
+      Magnum::Vector3{1, 1, 1},   Magnum::Vector3{1, -1, -1},
+      Magnum::Vector3{1, 1, -1},  Magnum::Vector3{1, -1, 1},
+      Magnum::Vector3{1, 1, -1},  Magnum::Vector3{-1, -1, -1},
+      Magnum::Vector3{-1, 1, -1}, Magnum::Vector3{1, -1, -1},
+      Magnum::Vector3{1, 1, 1},   Magnum::Vector3{-1, 1, -1},
+      Magnum::Vector3{-1, 1, 1},  Magnum::Vector3{1, 1, -1},
+      Magnum::Vector3{1, -1, 1},  Magnum::Vector3{-1, -1, -1},
+      Magnum::Vector3{-1, -1, 1}, Magnum::Vector3{1, -1, -1},
+      Magnum::Vector3{1, 1, 1},   Magnum::Vector3{-1, -1, 1},
+      Magnum::Vector3{-1, 1, 1},  Magnum::Vector3{1, -1, 1}};
```

Comment:
FYI -- the constructors are not explicit, so you can do just `{-1, 1, 1}` for example ;)